domain/consensus/processes/coinbasemanager/coinbasemanager.go

```diff
@@ -182,97 +179,47 @@ func acceptanceDataFromArrayToMap(acceptanceData externalapi.AcceptanceData) map
 // should have. This is mainly used for determining how much the coinbase for
 // newly generated blocks awards as well as validating the coinbase for blocks
 // has the expected value.
-//
-// Further details: https://hashdag.medium.com/kaspa-launch-plan-9a63f4d754a6
-func (c *coinbaseManager) CalcBlockSubsidy(blockHash *externalapi.DomainHash) (uint64, error) {
+func (c *coinbaseManager) CalcBlockSubsidy(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) (uint64, error) {
 	if blockHash.Equal(c.genesisHash) {
 		return c.subsidyGenesisReward, nil
 	}
-
-	return c.maxSubsidy, nil
-}
-
-func (c *coinbaseManager) calculateAveragePastSubsidy(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) (*big.Rat, error) {
-	const subsidyPastWindowSize = 100
-	blockWindow, err := c.dagTraversalManager.BlockWindow(stagingArea, blockHash, subsidyPastWindowSize)
+	blockDaaScore, err := c.daaBlocksStore.DAAScore(c.databaseContext, stagingArea, blockHash)
 	if err != nil {
-		return nil, err
+		return 0, err
 	}
-	if len(blockWindow) == 0 {
-		return new(big.Rat).SetFrac64(int64(c.subsidyGenesisReward), 1), nil
+	if blockDaaScore < c.deflationaryPhaseDaaScore {
+		return c.preDeflationaryPhaseBaseSubsidy, nil
 	}
 
-	pastBlocks, err := c.blockStore.Blocks(c.databaseContext, stagingArea, blockWindow)
-	if err != nil {
-		return nil, err
-	}
-
-	pastBlockSubsidySum := int64(0)
-	for _, pastBlock := range pastBlocks {
-		coinbaseTransaction := pastBlock.Transactions[transactionhelper.CoinbaseTransactionIndex]
-		_, _, pastBlockSubsidy, err := c.ExtractCoinbaseDataBlueScoreAndSubsidy(coinbaseTransaction)
-		if err != nil {
-			return nil, err
-		}
-		pastBlockSubsidySum += int64(pastBlockSubsidy)
-	}
-	return big.NewRat(pastBlockSubsidySum, int64(len(blockWindow))), nil
+	blockSubsidy := c.calcDeflationaryPeriodBlockSubsidy(blockDaaScore)
+	return blockSubsidy, nil
 }
 
-func (c *coinbaseManager) calculateMergeSetSubsidySum(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) (*big.Rat, error) {
-	ghostdagData, err := c.ghostdagDataStore.Get(c.databaseContext, stagingArea, blockHash, true)
-	if !database.IsNotFoundError(err) && err != nil {
-		return nil, err
-	}
-
-	// If there's ghostdag data with trusted data we prefer it because we need the original merge set non-pruned merge set.
-	if database.IsNotFoundError(err) {
-		ghostdagData, err = c.ghostdagDataStore.Get(c.databaseContext, stagingArea, blockHash, false)
-		if err != nil {
-			return nil, err
-		}
-	}
+func (c *coinbaseManager) calcDeflationaryPeriodBlockSubsidy(blockDaaScore uint64) uint64 {
+	const secondsPerMonth = 2629800
+	monthsSinceDeflationaryPhaseStarted := (blockDaaScore - c.deflationaryPhaseDaaScore) / secondsPerMonth
```

Comment:
I think we should avoid premature planning of BPS changes

Reply:
@someone235 what do you think?

Reply:
Anyway @stasatdaglabs, you need to be consistent. Because in the above line you didn't take the block rate into account, and a few lines under you did.

Reply:
Got rid of it for now